src/ORM/FieldType/DBDecimal.php

```diff
@@ -2,71 +2,55 @@
 
 namespace SilverStripe\ORM\FieldType;
 
+use SilverStripe\Forms\FormField;
 use SilverStripe\Forms\NumericField;
 use SilverStripe\ORM\DB;
+use SilverStripe\View\ViewableData;
 
 /**
  * Represents a Decimal field.
  */
 class DBDecimal extends DBField
 {
-
     /**
      * Whole number size
-     *
-     * @var int
      */
-    protected $wholeSize = 9;
+    protected int $wholeSize = 9;
 
     /**
      * Decimal scale
-     *
-     * @var int
      */
-    protected $decimalSize = 2;
+    protected int $decimalSize = 2;
 
     /**
      * Default value
-     *
-     * @var string
      */
-    protected $defaultValue = 0;
+    protected float|int|string $defaultValue = 0;
 
     /**
      * Create a new Decimal field.
-     *
-     * @param string $name
-     * @param int $wholeSize
-     * @param int $decimalSize
-     * @param float|int $defaultValue
      */
-    public function __construct($name = null, $wholeSize = 9, $decimalSize = 2, $defaultValue = 0)
+    public function __construct(?string $name = null, ?int $wholeSize = 9, ?int $decimalSize = 2, float|int $defaultValue = 0)
     {
         $this->wholeSize = is_int($wholeSize) ? $wholeSize : 9;
         $this->decimalSize = is_int($decimalSize) ? $decimalSize : 2;
 
-        $this->defaultValue = number_format((float) $defaultValue, $decimalSize ?? 0);
+        $this->defaultValue = number_format((float) $defaultValue, $this->decimalSize);
```

Comment:
Using 0 as a fallback was actually wrong - we should use the value we set for `$this->decimalSize`